kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java

```diff
@@ -165,7 +165,7 @@ protected void doStop() {
     if (metrics != null) {
       metrics.close();
     }
-    app.close(Duration.ofSeconds(30));
+    app.close(Duration.ofSeconds(150));
```

Comment:
@aman-bansal : You won't get 150 seconds afaik. K8s kills even before that I guess. You may want to check that once.
Can you please point me to a scenario where 30 seconds is not enough to shutdown?

cc: @ravisingal

Reply:
we need to increase `terminationGracePeriodSeconds` in deployments/statefulsets to more than 150 seconds.